packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -769,11 +806,28 @@ class _PageViewState extends State<PageView> {
     return null;
   }
 
+  double _getCacheExtent(AxisDirection direction, BoxConstraints constraints) {
+    // Use the actual child count if we can, otherwise use three for leading/active/trailing.
+    final int childCount = widget.childrenDelegate.estimatedChildCount ?? 3;
+    switch (direction) {
+      case AxisDirection.up:
+      case AxisDirection.down:
+        return constraints.maxHeight * childCount;
+      case AxisDirection.left:
+      case AxisDirection.right:
+        return constraints.maxWidth * childCount;
+    }
+    assert(false);
+    return null;
+  }
+
   @override
   Widget build(BuildContext context) {
     final AxisDirection axisDirection = _getDirection(context);
     final ScrollPhysics physics = widget.pageSnapping
-        ? _kPagePhysics.applyTo(widget.physics)
+        ? PageScrollPhysics(
+            allowImplicitScrolling: widget.allowImplicitScrolling,
+          ).applyTo(widget.physics)
```

Comment:
If page snapping is disabled, allowImplicitScrolling is ignored?

Reply:
It would result in cache extent being added but not necessarily the implicit scrolling, if the physics supplied (or platform default physics) dn't have it.